### PR TITLE
Cali dev

### PR DESCRIPTION
**Pull request by calizilla**
Hi, please test this set of benchmarking scripts on NCI Gadi. Please benchmark any multi-threading tool you have a script handy for. Ideally use small test data to make the test faster and use less SU. 

Please check that/comment on:

- [x] Instructions in README are clear and easy to follow
- [x] User is able to copy in commands to run a tool they want to benchmark, and easily set up the 'pbs' script to be compatible with the 'run' script using the 'outdir' and 'outfile_prefix' variables 
- [x] User can run benchmarks on different queues and CPU values
- [x] Output is as expected

### CODE REVIEW

**Comment by georgiesamaha** [comment]
@chlu0571 could you please make time to do this for Cali in the next week?

**Comment by chlu0571** [comment]
I tried the fastqc on the benchmarking script. it works fine in cali-dev version. since the resource is going to run out for er01, I will try others if we got more resources. The Readme fine is easy to follow.

File: README.md [hunk 1]
```diff
@@ -19,46 +21,81 @@ Currently, resource settings are defined for the following queues, being those t
...
+### 1. Rename the scripts
+
+Change the names of the pair of scripts, replacing `tool` with the name of the tool you are benchmarking.
+
+### 2. Add your tool commands to the benchmarking job script
+
+Open `<tool>_benchmark.pbs` with your preferred text editor, and perform the following edits:
+
+* Edit `-P` PBS directive to your NCI project
+* Edit `-lstorage` PBS directive to your required NCI storage paths, ensuring to use the [correct syntax for this directive](https://opus.nci.org.au/display/Help/PBS+Directives+Explained#PBSDirectivesExplained--lstorage=%3Cscratch/a00+gdata/xy11+massdata/a00%3E)
```
Comment (chlu0571) [inline README.md]: Edit '-l storage' PBS .....

**Review by chlu0571 — COMMENTED** [review]
The cali_dev version is definitely easier to operate, and the readme file also provides good instruction to perform the whole benchmarking process.

**Review by georgiesamaha — APPROVED** [review]

File: README.md [hunk 2]
```diff
@@ -1,12 +1,14 @@
...
+
+This is to benchmark resources for multi-threading tools, to help determine the optimal number of CPUs and queue type for the job on a single task before submtting the job on many tasks. Benchmarking in this way is critical to ensuring efficient and responsible use of HPC, saving time and compute costs in the long run. 
 
 ## Overview
 Given the need to frequently benchmark bioinformatics tools when developing pipelines, this repository contains a pair of sample scripts that can be adapted for use to other tools. 
 
 The template consists of a pair of scripts:
-* `<tool>_benchmark_run.sh` sets up the resources for each benchmark run
-* `<tool>_benchmark.pbs` is launched by the above, once for each identical analysis at the different resource thresholds
+* `<tool>_benchmark_run.sh` : sets up the resources for each benchmark run
```
Comment (georgiesamaha) [inline README.md]: Don't think this is needed.

File: README.md (hunk 1 above)
Comment (georgiesamaha) [reply README.md]: @chlu0571 what are you suggesting?

Comment (georgiesamaha) [inline README.md]:
```suggestion
* Edit `-l storage` PBS directive to your required NCI storage paths, ensuring to use the [correct syntax for this directive](https://opus.nci.org.au/display/Help/PBS+Directives+Explained#PBSDirectivesExplained--lstorage=%3Cscratch/a00+gdata/xy11+massdata/a00%3E)
```

**Review by georgiesamaha — DISMISSED** [review]
Added some comments and a suggestion.

File: README.md (hunk 1 above)
Comment (calizilla) [reply README.md]: Georgie added this change

File: README.md (hunk 2 above)
Comment (calizilla) [inline README.md]:
```suggestion
* `<tool>_benchmark_run.sh`  sets up the resources for each benchmark run
```

File: README.md [hunk 3]
```diff
@@ -1,12 +1,14 @@
...
+* `<tool>_benchmark.pbs` : is launched by the above, once for each identical analysis at the different resource thresholds
```
Comment (calizilla) [inline README.md]:
```suggestion
* `<tool>_benchmark.pbs`  is launched by the above, once for each identical analysis at the different resource thresholds
```

File: README.md (hunk 2 above)
Comment (calizilla) [reply README.md]: Fixed - [d99f7d3](https://github.com/Sydney-Informatics-Hub/Gadi-benchmarking/pull/1/commits/d99f7d3c75a55c261386ecaf41b8f79116c884cf) and [9dbbf76](https://github.com/Sydney-Informatics-Hub/Gadi-benchmarking/pull/1/commits/9dbbf760048d1be3bcf7959cc85e6c731d8a0c89)

**Review by georgiesamaha — APPROVED** [review]
Looks great!

**Pull request merged**